j5/backends/hardware/sr/v4/serial/power_board.py

```diff
@@ -172,18 +178,17 @@ def buzz(
                              f"the only valid identifier is 0.")
 
         duration_ms = round(duration.total_seconds() * 1000)
-        max_duration = (2 ** 31) - 1  # int32 max
-        if duration_ms > max_duration:
+
+        if duration_ms not in range(MAX_BUZZ_DURATION_MS + 1):
```

Comment:
This seems horribly inefficient. The only improvement the original line needed was to check that duration wasn't negative.

Reply:
Sorry, I'm not sure what you mean by this being inefficient?

Reply:
At first glance here it appears that you are about to do 2 billion iterations to find out if this generator contains your value. It does appear that range impliments a contains method to shortcut this (still ~7x slower than the straight compare). However the exclusive top end of this range makes it arguably less readable than simply comparing it's bounds.

Reply:
Interesting that it's that much slower, usually using a range object is just as fast (There's a bytecode op for it)